app/lib/world/GoalManager.coffee

```diff
@@ -96,6 +105,40 @@ module.exports = class GoalManager extends CocoClass
       @goalStates[goalID] = goalState
     @notifyGoalChanges()
 
+  # Adds any goals for the current capstoneStage
+  # Returns the current capstoneStage
+  addAdditionalGoals: (session, additionalGoals) ->
+    state = session.get('state') ? {}
+    if not state.capstoneStage
+      # In daily speak, we think of initial goals as stage 1 and additional goals
+      # as stage 2 and above. That is why we are starting from 2.
+      state.capstoneStage = 2
```

Comment:
It seems we are thinking of capstoneStage as the current stage and not the stage completed(I was assuming this to be stage completed until now :) ).
So in this case, the stage completed = stage.capstoneStage - 1, is that right? (Want to clarify since the unit map next level logic will depend on this)

Reply:
Also this means that at the end of the capstone level's last stage, capstoneStage will increment to a value for which we ideally don't have any goals, but `capstoneStage - 1 = level's last stage`, right?
Can we add this as a comment somewhere here to make it clear.